src/proxy/inbound.rs

```diff
@@ -259,138 +291,170 @@ impl Inbound {
+            // which drops the requirement of spoofing IPs from waypoints
+            super::get_original_src_from_fwded(&req).unwrap_or(conn.src_ip)
+        } else {
+            conn.src_ip
+        };
 
-                Ok(Response::builder()
-                    .status(status_code)
-                    .body(Empty::new())
-                    .unwrap())
+        let baggage =
+            parse_baggage_header(req.headers().get_all(BAGGAGE_HEADER)).unwrap_or_default();
+
+        let source = match from_gateway {
+            true => None, // we cannot lookup source workload since we don't know the network, see https://github.com/istio/ztunnel/issues/515
+            false => {
+                let src_network_addr = NetworkAddress {
+                    // we can assume source network is our network because we did not traverse a gateway
+                    network: conn.dst_network.to_string(),
+                    address: source_ip,
+                };
+                // Find source info. We can lookup by XDS or from connection attributes
+                state.fetch_workload(&src_network_addr).await
             }
-            // Return the 404 Not Found for other routes.
-            method => {
-                info!("Sending 404, got {method}");
-                Ok(Response::builder()
-                    .status(StatusCode::NOT_FOUND)
-                    .body(Empty::new())
-                    .unwrap())
+        };
+
+        let derived_source = metrics::DerivedWorkload {
+            identity: conn.src_identity.clone(),
+            cluster_id: baggage.cluster_id,
+            namespace: baggage.namespace,
+            workload_name: baggage.workload_name,
+            revision: baggage.revision,
+            ..Default::default()
+        };
+
+        let (upstream, ds) = match &wl_svcs {
+            Some((upstream, svcs)) => {
+                let svc = proxy::guess_inbound_service(&conn, svcs.clone(), upstream);
+                (Some(upstream.clone()), svc)
             }
-        }
+            // TODO should upstream be something that repr the Waypoint we're fwding to?
+            _ => (None, None),
+        };
+
+        let connection_metrics = ConnectionOpen {
+            reporter: Reporter::destination,
+            source,
+            derived_source: Some(derived_source),
+            destination: upstream,
+            connection_security_policy: metrics::SecurityPolicy::mutual_tls,
+            destination_service: ds,
+        };
+
+        let (next_hop, inbound_req) = match sandwich {
+            false => (addr, Hbone(req)),
+            true => (
+                conn.dst,
+                Sandwich(
+                    // TODO add src_port or change conn.src_ip to src (SocketAddr)
+                    req,
+                    (SocketAddr::new(conn.src_ip, 10000), addr),
+                    conn.src_identity.unwrap().to_string(),
```

Comment:
Is it possible to reach this point with no src identity? I don't think so but not 100% sure.

Reply:
Yeah, definitely need to cleanup unwraps although I'm pretty sure only auth'd conns make it here.